src/NzbDrone.Common.Test/Http/HappyEyeballsFixture.cs

```diff
@@ -0,0 +1,222 @@
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using NzbDrone.Common.Http.HappyEyeballs;
+using NzbDrone.Test.Common;
+
+namespace NzbDrone.Common.Test.Http
+{
+    [TestFixture]
+    public class HappyEyeballsFixture : TestBase
+    {
+        private static IPAddress ipv6Address1 = IPAddress.Parse("2001:db8::1");
+        private static IPAddress ipv6Address2 = IPAddress.Parse("2001:db8::2");
+        private static IPAddress ipv4Address1 = IPAddress.Parse("192.0.2.1");
+        private static IPAddress ipv4Address2 = IPAddress.Parse("192.0.2.2");
+
+        private Mock<Func<IPAddress, CancellationToken, Task<IDisposable>>> _connectSocketMock;
+        private Mock<Func<CancellationToken, Task>> _taskDelayMock;
+        private HappyEyeballs<IDisposable> _happyEyeballs;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _connectSocketMock = new Mock<Func<IPAddress, CancellationToken, Task<IDisposable>>>(MockBehavior.Strict);
+            _taskDelayMock = new Mock<Func<CancellationToken, Task>>(MockBehavior.Strict);
+            _happyEyeballs = new HappyEyeballs<IDisposable>(_connectSocketMock.Object, _taskDelayMock.Object);
+        }
+
+        [Test]
+        public void should_throw_exception_when_no_ips_resolved()
+        {
+            var addresses = Array.Empty<IPAddress>();
+            var cancellationToken = CancellationToken.None;
+
+            Assert.ThrowsAsync<ArgumentOutOfRangeException>(async () =>
+                await _happyEyeballs.Connect(addresses, cancellationToken));
+        }
+
+        [Test]
+        public async Task should_connect_successfully_when_valid_addresses_are_provided()
+        {
+            var addresses = new[]
+            {
+                ipv4Address1,
+                ipv6Address1,
+            };
+
+            var cancellationToken = CancellationToken.None;
+
+            var sequence = new MockSequence();
+            _connectSocketMock.InSequence(sequence)
+                .Setup(x => x(ipv6Address1, It.IsAny<CancellationToken>()))
+                .ReturnsAsync(Mock.Of<IDisposable>(MockBehavior.Strict));
+            _taskDelayMock.InSequence(sequence)
+                .Setup(x => x(It.IsAny<CancellationToken>()))
+                .Returns(TaskFromCancellationToken);
+
+            var result = await _happyEyeballs.Connect(addresses, cancellationToken);
+
+            Assert.NotNull(result);
```

Comment:
Use `FluentAssertions` instead:

```suggestion
            result.Should().NotBeNull();
```